src/ecs/resources/input.rs

```diff
@@ -404,25 +359,27 @@ impl InputHandler {
     // can't think of a use case for it.
 
     /// Checks if the given button is currently pressed.
-    pub fn action_is_pressed(&self, action: i32) -> Option<bool> {
+    pub fn action_is_pressed<T: AsRef<str>>(&self, action: T) -> Option<bool> {
         self.actions
-            .get(&action)
-            .map(|&b| self.button_is_pressed(b))
+            .get(action.as_ref())
+            .map(|ref buttons| buttons.iter().any(|&b| self.button_is_pressed(b)))
     }
 
     /// Checks if all the given actions are pressed.
     ///
     /// If any action in this list is invalid this will return the id of it in Err.
-    pub fn actions_are_pressed(&self, actions: &[i32]) -> Result<bool, Vec<i32>> {
+    pub fn actions_are_pressed<T: AsRef<str>>(&self, actions: &[T]) -> Result<bool, Vec<String>> {
         let mut all_buttons_are_pressed = true;
         let mut bad_values = Vec::new();
         for action in actions {
```

Comment:
Could you use iterators to make this code nicer? I don't think there is a need to report multiple ill-named actions at once, given that is pretty rare, so that could be another simplification.